asllib/desugar.ml

```diff
@@ -53,8 +53,17 @@ let desugar_setter call fields rhs =
       (* Setter(rhs, ...); *)
       let write =
         let temp_e = E_Var temp |> here in
-        S_Call
-          { name; args = temp_e :: args; params = []; call_type = ST_Setter }
+        S_Call { name; args = temp_e :: args; params; call_type = ST_Setter }
         |> here
       in
       S_Seq (s_then read modify, write)
+
+let desugar_elided_parameter ldk lhs (call : call annotated) =
+  let bits_e =
+    match lhs with
+    | LDI_Typed (_, { desc = T_Bits (bits_e, []) }) -> bits_e
+    | _ -> Error.fatal_from (to_pos call) CannotParse
```

Comment:
I think this is more of a parse error than obsolete syntax, in that it should be illegal to write e.g. `let x = foo{,M}(args);` because there is no way to desugar this to insert the elided parameter. However the old syntax omitting the parameters entirely is still legal syntactically (but not for type checking). I could add a desugaring error if you'd like?